### PR TITLE
Update documentation of Java Tracker according to recent version

### PR DESCRIPTION
**Pull request by dheid**
Hi @sgiehl,

I updated the Matomo Java Tracker according to your review comments and fixed some mistakes (wrong Javadoc URL, spelling errors). 

Thanks again

Daniel

### CODE REVIEW

File: docs/4.x/tracking-java.md [hunk 1]
```diff
@@ -31,60 +43,78 @@ implementation 'org.piwik.java.tracking:matomo-java-tracker:3.x.x'
...
-In this example the tracker will send the tracking requests to the Matomo server at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all other configuration properties.
+In this example the tracker will send the tracking requests to the Matomo server
+at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all
+other configuration properties.
 
 ## MatomoTracker
 
-The class `MatomoTracker` is the main class to use when sending tracking requests. It contains methods to send tracking requests synchronously and asynchronously.
+The
+class [`MatomoTracker`](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/org/matomo/java/tracking/MatomoTracker.html)
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
The class [`MatomoTracker`](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/org/matomo/java/tracking/MatomoTracker.html)
```

File: docs/4.x/tracking-java.md [hunk 2]
```diff
@@ -31,60 +43,78 @@ implementation 'org.piwik.java.tracking:matomo-java-tracker:3.x.x'
...
+is the main class to use for sending tracking requests. It contains methods to send tracking
+requests synchronously and asynchronously.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
is the main class to use for sending tracking requests. It contains methods to send tracking requests synchronously and asynchronously.
```

File: docs/4.x/tracking-java.md [hunk 3]
```diff
@@ -31,60 +43,78 @@ implementation 'org.piwik.java.tracking:matomo-java-tracker:3.x.x'
...
+TrackerConfiguration configuration=TrackerConfiguration.builder()
         .apiEndpoint(URI.create("https://your-domain.net/matomo/matomo.php"))
         .defaultSiteId(42) // if not explicitly specified by the request
         .build();
 ```
 
-In this example the tracker will send the tracking requests to the Matomo server at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all other configuration properties.
+In this example the tracker will send the tracking requests to the Matomo server
+at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all
+other configuration properties.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
In this example the tracker will send the tracking requests to the Matomo server at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all other configuration properties.
```

File: docs/4.x/tracking-java.md [hunk 4]
```diff
@@ -31,60 +43,78 @@ implementation 'org.piwik.java.tracking:matomo-java-tracker:3.x.x'
...
-The class `TrackerConfiguration` configures the tracker and can be built in a fluent way. See the configuration properties in the [JavaDoc of TrackerConfiguration](https://matomo-org.github.io/matomo-java-tracker/javadoc/org/matomo/java/tracking/TrackerConfiguration.html).
+The class `TrackerConfiguration` configures the tracker and can be built in a fluent way. See the configuration
+properties in
+the [JavaDoc of TrackerConfiguration](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/org/matomo/java/tracking/TrackerConfiguration.html).
 
 To create a tracker configuration fluently you can use the `TrackerConfiguration.builder()` method:
 
 ```java
-TrackerConfiguration configuration = TrackerConfiguration.builder()
+TrackerConfiguration configuration=TrackerConfiguration.builder()
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
TrackerConfiguration configuration = TrackerConfiguration.builder()
```

File: docs/4.x/tracking-java.md [hunk 5]
```diff
@@ -1,25 +1,37 @@
...
-The primary source to look for an API documentation is the [JavaDoc](https://matomo-org.github.io/matomo-java-tracker/javadoc/index.html). This guide provides a quick overview of the most important classes and methods of the library.
+For detailed API documentation, refer to the [JavaDoc](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/index.html). This
+documentation provides a swift and informative
+overview of the essential classes and methods within the library.
 
-If you experience any problems or have any questions, please [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). We are happy to help you.
+If you encounter any challenges or have inquiries, feel free to [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in
+the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). Our team is dedicated to
+assisting you promptly and ensuring a smooth experience with the Matomo Java
+Tracker.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
assisting you promptly and ensuring a smooth experience with the Matomo Java Tracker.
```

File: docs/4.x/tracking-java.md [hunk 6]
```diff
@@ -1,25 +1,37 @@
...
 category: API Reference
 ---
+
 # Java Tracker
 
-The Matomo Java Tracker is a Java library to send tracking data to Matomo servers. It is compatible with Java 1.8 and above. You can use it to track data from any Java application, including Java web applications, Java desktop applications, Java command line applications, Java batch applications, Java microservices and more.
+The Matomo Java Tracker is a comprehensive Java library designed for sending tracking data to Matomo servers.
+It is compatible with Java 1.8 and later versions. Whether you're working with Java web applications, desktop
+applications, command line applications, batch
+applications, microservices, or other Java-based projects, the Matomo Java Tracker has you covered.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]: Were the lines split to multiple lines by some tool or did you do it on purpose?
I don't think we have a line length limit we need to follow and some of the lines are unnecessarily too short for me.

Comment (michalkleiner) [reply docs/4.x/tracking-java.md]: It should be fine to have 120 characters line length in the .md files.

Comment (dheid) [reply docs/4.x/tracking-java.md]: My IDE did that. I deactivated the formatter, but somehow it destroys my formatting everytime.

**Comment by dheid** [comment]
@michalkleiner Thanks for your review! I (manually) fixed the line breaks (hopefully). That was exhausting. Is it better now?

**Comment by michalkleiner** [comment]
Thanks @dheid, I'll have another look tomorrow. The formatting/line breaks were adding too much noise to the actual content changes so it was making it harder to review (nothing personal there!) Thanks for the tweak.

**Comment by dheid** [comment]
@michalkleiner Any updates?

**Comment by dheid** [comment]
@sgiehl Could you have a look at it, please?

File: docs/4.x/tracking-java.md [hunk 7]
```diff
@@ -1,25 +1,35 @@
...
 
-The primary source to look for an API documentation is the [JavaDoc](https://matomo-org.github.io/matomo-java-tracker/javadoc/index.html). This guide provides a quick overview of the most important classes and methods of the library.
+For detailed API documentation, refer to the [JavaDoc](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/index.html).
+This documentation provides a swift and informative overview of the essential classes and methods within the library.
 
-If you experience any problems or have any questions, please [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). We are happy to help you.
+If you encounter any challenges or have inquiries, feel free to [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in
+the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). Our team is dedicated to
+assisting you promptly and ensuring a smooth experience with the Matomo Java Tracker.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). Our team will be happy to help.
```

File: docs/5.x/tracking-java.md [hunk 8]
```diff
@@ -1,25 +1,35 @@
...
 
-The primary source to look for an API documentation is the [JavaDoc](https://matomo-org.github.io/matomo-java-tracker/javadoc/index.html). This guide provides a quick overview of the most important classes and methods of the library.
+For detailed API documentation, refer to the [JavaDoc](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/index.html).
+This documentation provides a swift and informative overview of the essential classes and methods within the library.
 
-If you experience any problems or have any questions, please [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). We are happy to help you.
+If you encounter any challenges or have inquiries, feel free to [create an issue](https://github.com/matomo-org/matomo-java-tracker/issues/new) in
+the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). Our team is dedicated to
+assisting you promptly and ensuring a smooth experience with the Matomo Java Tracker.
```
Comment (michalkleiner) [inline docs/5.x/tracking-java.md]:
```suggestion
the [Matomo Java Tracker repository](https://github.com/matomo-org/matomo-java-tracker). Our team will be happy to help.
```

File: docs/4.x/tracking-java.md [hunk 9]
```diff
@@ -42,72 +53,74 @@ TrackerConfiguration configuration = TrackerConfiguration.builder()
...
-In this example the tracker will send the tracking requests to the Matomo server at `https://your-domain.net/matomo/matomo.php` and use the site id `42`. The tracker will use the default values for all other configuration properties.
+In this example the tracker will send the tracking requests to the Matomo server at `https://your-domain.net/matomo/matomo.php`
+and use the site id `42`. The tracker will use the default values for all other configuration properties.
 
 ## MatomoTracker
 
-The class `MatomoTracker` is the main class to use when sending tracking requests. It contains methods to send tracking requests synchronously and asynchronously.
+The class [`MatomoTracker`](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker-core/latest/org/matomo/java/tracking/MatomoTracker.html)
+is the main class to use for sending tracking requests. It contains methods to send tracking requests synchronously and
+asynchronously.
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
is the main class to use for sending tracking requests. It contains methods to send tracking requests synchronously and asynchronously.
```

File: docs/4.x/tracking-java.md [hunk 10]
```diff
@@ -120,85 +133,247 @@ TrackerConfiguration configuration = TrackerConfiguration.builder()
...
+the constructor to send the requests via HTTP POST. Using the method `CompletableFuture.join()` you can wait for the request to
+be sent. If the request fails, the method will throw an unchecked exception.
 
-## Authentikation Token
+## Authentication Token
 
-The parameter `token_auth` contains the authentication token of the Matomo user. Some parameters require it. The authorization key can be found in the Matomo user interface under "Administration" -> "Personal" -> "Security" -> "Auth tokens"
+The `token_auth` parameter contains the authentication token of the Matomo user. Some parameters require it. The
+authorization key can be found in the Matomo user interface under "Administration" -> "Personal" -> "Security" -> "Auth
+tokens"
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
The authorization key can be found in the Matomo user interface under "Administration" -> "Personal" -> "Security" -> "Auth tokens"
```

File: docs/4.x/tracking-java.md [hunk 11]
```diff
@@ -120,85 +133,247 @@ TrackerConfiguration configuration = TrackerConfiguration.builder()
...
+name and action URL. It also sets a custom visitor id.
+The tracker will send the request asynchronously and return immediately. The tracker will use the configuration set in
+the constructor to send the requests via HTTP POST. Using the method `CompletableFuture.join()` you can wait for the request to
+be sent. If the request fails, the method will throw an unchecked exception.
 
-## Authentikation Token
+## Authentication Token
 
-The parameter `token_auth` contains the authentication token of the Matomo user. Some parameters require it. The authorization key can be found in the Matomo user interface under "Administration" -> "Personal" -> "Security" -> "Auth tokens"
+The `token_auth` parameter contains the authentication token of the Matomo user. Some parameters require it. The
```
Comment (michalkleiner) [inline docs/4.x/tracking-java.md]:
```suggestion
The `token_auth` parameter contains the authentication token of the Matomo user. Some parameters require it.
```

**Comment by michalkleiner** [comment]
Still more unnecessary line breaks than I expected, leaving stray "The" articles at the end of lines etc. Probably needs another pass and really focus on not changing stuff that doesn't need changing in terms of content.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**